Wikipedia/Code/TalkPageViewController+FindInPage.swift

```diff
@@ -0,0 +1,104 @@
+import Foundation
+
+extension TalkPageViewController {
+
+    // MARK: - Overrides
+
+    override var canBecomeFirstResponder: Bool {
+        return findInPageState.keyboardBar != nil
+    }
+
+    override var inputAccessoryView: UIView? {
+        return findInPageState.keyboardBar
+    }
+
+    // MARK: - Presentation
+
+    fileprivate func createFindInPageViewIfNecessary() {
+        guard findInPageState.keyboardBar == nil else {
+            return
+        }
+
+        let keyboardBar = FindAndReplaceKeyboardBar.wmf_viewFromClassNib()!
+        keyboardBar.delegate = self
+        keyboardBar.apply(theme: theme)
+        findInPageState.keyboardBar = keyboardBar
+    }
+
+    func showFindInPage() {
+        createFindInPageViewIfNecessary()
+        becomeFirstResponder()
+        findInPageState.keyboardBar?.show()
+    }
+
+    func hideFindInPage() {
+        findInPageState.reset(viewModel.topics)
+        findInPageState.keyboardBar?.hide()
+        talkPageView.collectionView.reloadData()
+        resignFirstResponder()
+    }
+
+    // MARK: - Scroll to Element
+
+    func scrollToFindInPageResult(_ result: TalkPageFindInPageSearchController.SearchResult?) {
+        guard let result = result else { return }
+
+        talkPageView.collectionView.reloadData()
+
+        // TODO: - Fine tune position when scrolling
+        switch result.location {
+        case .topicTitle(topicIndex: let index, topicIdentifier: _):
+            let indexPath = IndexPath(row: index, section: 0)
+            talkPageView.collectionView.scrollToItem(at: indexPath, at: .top, animated: true)
+        case .topicLeadComment(topicIndex: let index, replyIdentifier: _):
+            let commentViewModel = viewModel.topics[index].leadComment
+            scrollToComment(commentViewModel: commentViewModel, animated: true)
+        case .reply(topicIndex: let topicIndex, topicIdentifier: _, replyIndex: let replyIndex, replyIdentifier: _):
+            let topicViewModel = viewModel.topics[topicIndex]
+            topicViewModel.isThreadExpanded = true
```

Comment:
Okay, actually you might need a reloadData call after this for the cell to expand 😅 I think it might be more simple at this point to just open all the threads (however the overflow menu button is working) as soon as the Find in Page is button is tapped instead of disclosing them as they navigate through the results, since `reloadData()` here seems to mess with the scrolling.

Reply:
Got it – updated to open all threads when find in page is first tapped. Personally, I think this behavior of expanding _all_ threads when entering find in page is a little heavy, but it's nice to have better scrolling. In a future improvement to this, I think it might be better if we can move this thread expansion earlier in the process after the user has completed a search and keep the behavior to only opening the threads with matches.

Reply:
> In a future improvement to this, I think it might be better if we can move this thread expansion earlier in the process after the user has completed a search and keep the behavior to only opening the threads with matches.

Definitely agreed!